sdk/ai/azure-ai-inference/azure/ai/inference/_patch.py

```diff
@@ -0,0 +1,1037 @@
+            body = {
+                "frequency_penalty": frequency_penalty,
+                "max_tokens": max_tokens,
+                "messages": messages,
+                "presence_penalty": presence_penalty,
+                "response_format": response_format,
+                "seed": seed,
+                "stop": stop,
+                "stream": stream,
+                "temperature": temperature,
+                "tool_choice": tool_choice,
+                "tools": tools,
+                "top_p": top_p,
+            }
+            if model_extras is not None and bool(model_extras):
+                body.update(model_extras)
+                _unknown_params = _models._enums.UnknownParams.PASS_THROUGH  # pylint: disable=protected-access
+            body = {k: v for k, v in body.items() if v is not None}
+        elif isinstance(body, dict) and "stream" in body and isinstance(body["stream"], bool):
+            stream = body["stream"]
+        content_type = content_type or "application/json"
+        _content = None
+        if isinstance(body, (IOBase, bytes)):
+            _content = body
+        else:
+            _content = json.dumps(body, cls=SdkJSONEncoder, exclude_readonly=True)  # type: ignore
+
+        _request = build_chat_completions_complete_request(
+            unknown_params=_unknown_params,
+            content_type=content_type,
+            api_version=self._config.api_version,
+            content=_content,
+            headers=_headers,
+            params=_params,
+        )
+        path_format_arguments = {
+            "endpoint": self._serialize.url("self._config.endpoint", self._config.endpoint, "str", skip_quote=True),
+        }
+        _request.url = self._client.format_url(_request.url, **path_format_arguments)
+
+        _stream = stream or False
+        pipeline_response: PipelineResponse = self._client._pipeline.run(  # pylint: disable=protected-access
+            _request, stream=_stream, **kwargs
+        )
+
+        response = pipeline_response.http_response
+
+        if response.status_code not in [200]:
+            if _stream:
+                response.read()  # Load the body in memory and close the socket
+            map_error(status_code=response.status_code, response=response, error_map=error_map)
+            raise HttpResponseError(response=response)
+
+        if _stream:
+            return _models.StreamingChatCompletions(response)
+
+        return _deserialize(_models._models.ChatCompletions, response.json())  # pylint: disable=protected-access
+
+
+    @distributed_trace
+    def get_model_info(self, **kwargs: Any) -> _models.ModelInfo:
+        # pylint: disable=line-too-long
+        """Returns information about the AI model.
+
+        :return: ModelInfo. The ModelInfo is compatible with MutableMapping
+        :rtype: ~azure.ai.inference.models.ModelInfo
+        :raises ~azure.core.exceptions.HttpResponseError
+        """
+        if not hasattr(self, "_model_info"):
```

Comment:
The canonical way to do this is to initialize self._model_info as None in the __init__.

Reply:
(or, if you want to be fancy, https://docs.python.org/3/library/functools.html#functools.cached_property)

Reply:
Yes, but that would require me to override the __init__ method of all the clients, sync and async flavors. That's a bit too risky for me at this late stage. We can consider this again in the next release.

Reply:
`cached_property` is neat though!

Reply:
Changed my mind! fixed with commit https://github.com/Azure/azure-sdk-for-python/pull/34937/commits/864796470f52b50be43883916f5b1360ca521612